tools/depends/target/boost/Makefile

```diff
@@ -15,23 +15,15 @@ $(TARBALLS_LOCATION)/$(ARCHIVE):
 $(PLATFORM): $(TARBALLS_LOCATION)/$(ARCHIVE) $(DEPS)
 	rm -rf $(PLATFORM); mkdir -p $(PLATFORM)
 	cd $(PLATFORM); $(ARCHIVE_TOOL) $(ARCHIVE_TOOL_FLAGS) $(TARBALLS_LOCATION)/$(ARCHIVE)
-	echo 'using gcc : : $(CC) : ' > $(PLATFORM)/tools/build/v2/user-config.jam
-	echo '	<cflags>"$(CFLAGS)"' >> $(PLATFORM)/tools/build/v2/user-config.jam
-	echo '  <cxxflags>"$(CXXFLAGS)"' >> $(PLATFORM)/tools/build/v2/user-config.jam
-	echo '	;' >> $(PLATFORM)/tools/build/v2/user-config.jam
-	cd $(PLATFORM); patch -p1 < ../add-arm-mem-barrier.patch
-	cd $(PLATFORM); patch -p1 < ../fix-deprecated-swp.patch
 ifeq (arm, $(findstring arm, $(CPU)))
-	cd $(PLATFORM); patch -p0 < ../fix_shared_ptr_multithread.patch
+	#ugly hack to avoid issue with line endings. boost headers use \r\n and patch
+	#will be converted by git to \n and fail to apply
+	cd $(PLATFORM); cat ../fix_shared_ptr_multithread.patch >> boost/config/user.hpp
```

Comment:
@Memphiz @wsnipex I welcome any suggestions on how to solve this in prettier way.